### PR TITLE
Update Jbake version to support Java 11

### PR DESCRIPTION
**Pull request by hectorespert**
Jbake doesn't work with Java 9 or higher: https://github.com/jbake-org/jbake/issues/408

Based on https://github.com/apache/netbeans-website/pull/431

### CODE REVIEW

**Review by vieiro — CHANGES_REQUESTED** [review]
Hi @hectorespert ,

Thanks for this long-due PR. 

It seems that this fails when running with Java 8 _and_ Java 11 with [1]. I've run it with `.\gradlew clean buildSite run` on Windows 10.

Would it be possible to keep backwards compatibility with Java 8? Or are we dropping it?

I'm asking because the Jenkins job responsible for building the site (https://ci-builds.apache.org/job/Netbeans/job/netbeans-website/) requires a Java 8 version (in the Jenkins job, see `tools { jdk 'jdk_1.8_latest' }` ). So we should modify that before merging or keep backwards compatibility with Java 8.

Thanks,
Antonio


[1]
```
<stdin>: line 500: no callout found for <1>
$ANSI{green {db=cache}} Error during comparision
java.lang.ClassCastException: java.util.Date cannot be cast to java.lang.String
        at java.lang.String.compareTo(String.java:111)
        at com.orientechnologies.orient.core.sql.parser.OOrderByItem.compare(OOrderByItem.java:137)
        at com.orientechnologies.orient.core.sql.parser.OOrderBy.compare(OOrderBy.java:58)
        at com.orientechnologies.orient.core.sql.executor.OrderByStep.lambda$init$3(OrderByStep.java:150)
        at java.util.TimSort.countRunAndMakeAscending(TimSort.java:360)
        at java.util.TimSort.sort(TimSort.java:234)
        at java.util.Arrays.sort(Arrays.java:1512)
        at java.util.ArrayList.sort(ArrayList.java:1464)
        at com.orientechnologies.orient.core.sql.executor.OrderByStep.init(OrderByStep.java:150)
        at com.orientechnologies.orient.core.sql.executor.OrderByStep.lambda$syncPull$0(OrderByStep.java:40)
        at java.util.Optional.ifPresent(Optional.java:159)
        at com.orientechnologies.orient.core.sql.executor.OrderByStep.syncPull(OrderByStep.java:40)
        at com.orientechnologies.orient.core.sql.executor.OSelectExecutionPlan.fetchNext(OSelectExecutionPlan.java:37)
        at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.fetchNext(OLocalResultSet.java:39)
        at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.<init>(OLocalResultSet.java:30)
        at com.orientechnologies.orient.core.sql.parser.OSelectStatement.execute(OSelectStatement.java:276)
        at com.orientechnologies.orient.core.sql.parser.OStatement.execute(OStatement.java:79)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.query(ODatabaseDocumentEmbedded.java:538)
        at org.jbake.app.ContentStore.query(ContentStore.java:339)
        at org.jbake.app.ContentStore.getUnrenderedContent(ContentStore.java:311)
        at org.jbake.render.DocumentsRenderer.render(DocumentsRenderer.java:25)
        at org.jbake.app.Oven.renderContent(Oven.java:207)
        at org.jbake.app.Oven.bake(Oven.java:156)
        at org.jbake.app.Oven$bake$2.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:119)
        at org.jbake.gradle.impl.JBakeWorkAction.execute(JBakeWorkAction.groovy:35)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
        at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:47)
        at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:41)
        at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:98)
        at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:41)
        at org.gradle.workers.internal.IsolatedClassloaderWorker.execute(IsolatedClassloaderWorker.java:49)
        at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.lambda$execute$0(IsolatedClassloaderWorkerFactory.java:57)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
        at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.execute(IsolatedClassloaderWorkerFactory.java:49)
        at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:198)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:215)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by vieiro** [comment]
In java 11 the error I'm getting is slightly different:

```
 class java.util.Date cannot be cast to class java.lang.String (java.util.Date and java.lang.String are in module java.base of loader 'bootstrap')
```

**Comment by vieiro** [comment]
Hi @hectorespert ,

I still see a bug [1] when rendering on both Windows and Linux. If were to merge then this would break the Jenkins job (and the rendering of the site).

Any ideas of what could be the cause? I'll try to investigate as well.


```
<stdin>: line 500: no callout found for <1>
<stdin>: line 470: list item index: expected 1, got 6
$ANSI{green {db=cache}} Error during comparision
java.lang.ClassCastException: java.util.Date cannot be cast to java.lang.String
        at java.lang.String.compareTo(String.java:111)
        at com.orientechnologies.orient.core.sql.parser.OOrderByItem.compare(OOrderByItem.java:137)
        at com.orientechnologies.orient.core.sql.parser.OOrderBy.compare(OOrderBy.java:58)
        at com.orientechnologies.orient.core.sql.executor.OrderByStep.lambda$init$3(OrderByStep.java:150)
        at java.util.TimSort.countRunAndMakeAscending(TimSort.java:360)
        at java.util.TimSort.sort(TimSort.java:234)
        at java.util.Arrays.sort(Arrays.java:1512)
        at java.util.ArrayList.sort(ArrayList.java:1464)
        at com.orientechnologies.orient.core.sql.executor.OrderByStep.init(OrderByStep.java:150)
        at com.orientechnologies.orient.core.sql.executor.OrderByStep.lambda$syncPull$0(OrderByStep.java:40)
        at java.util.Optional.ifPresent(Optional.java:159)
        at com.orientechnologies.orient.core.sql.executor.OrderByStep.syncPull(OrderByStep.java:40)
        at com.orientechnologies.orient.core.sql.executor.OSelectExecutionPlan.fetchNext(OSelectExecutionPlan.java:37)
        at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.fetchNext(OLocalResultSet.java:39)
        at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.<init>(OLocalResultSet.java:30)
        at com.orientechnologies.orient.core.sql.parser.OSelectStatement.execute(OSelectStatement.java:276)
        at com.orientechnologies.orient.core.sql.parser.OStatement.execute(OStatement.java:79)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.query(ODatabaseDocumentEmbedded.java:538)
        at org.jbake.app.ContentStore.query(ContentStore.java:339)
        at org.jbake.app.ContentStore.getUnrenderedContent(ContentStore.java:311)
        at org.jbake.render.DocumentsRenderer.render(DocumentsRenderer.java:25)
        at org.jbake.app.Oven.renderContent(Oven.java:207)
        at org.jbake.app.Oven.bake(Oven.java:156)
        at org.jbake.app.Oven$bake$2.call(Unknown Source)
```

**Comment by hectorespert** [comment]
> Any ideas of what could be the cause? I'll try to investigate as well.

Any idea.

File: netbeans.apache.org/src/content/wiki/BuildSystemHowTo.asciidoc [hunk 1]
```diff
@@ -30,7 +30,6 @@
 :toc: left
 :toc-title:
 :experimental:
-:date: June 21, 2006
```
Comment (hectorespert) [inline netbeans.apache.org/src/content/wiki/BuildSystemHowTo.asciidoc:33]: @vieiro Seems that date exceptions are generated by this document metadata.

**Review by vieiro — APPROVED** [review]
Thanks, @hectorespert . Things run smoothly now! Feel free to merge.

**Pull request merged**